lib/segment/src/data_types/named_vectors.rs

```diff
@@ -74,35 +165,62 @@ impl<'a> NamedVectors<'a> {
         self.map.iter().map(|(k, _)| k.as_ref())
     }
 
-    pub fn into_default_vector(mut self) -> Option<Vec<VectorElementType>> {
-        self.map
-            .get_mut(DEFAULT_VECTOR_NAME)
-            .map(|src| std::mem::take(src).into_owned())
+    pub fn into_default_vector(mut self) -> Option<Vector> {
+        self.map.get_mut(DEFAULT_VECTOR_NAME).map(|src| match src {
+            CowValue::Dense(src) => Vector::Dense(std::mem::take(src).into_owned()),
+            CowValue::Sparse(src) => Vector::Sparse(std::mem::take(src).into_owned()),
```

Comment:
In practice we do not want to support anonymous sparse vector.
Is this conversion mandatory?

Reply:
marked as todo to remove this functions